packages/amplify-category-api/src/provider-utils/awscloudformation/ecs-stack.ts

```diff
@@ -0,0 +1,426 @@
+      },
+    });
+
+    new apigw2.CfnStage(this, 'Stage', {
+      apiId: cdk.Fn.ref(api.logicalId),
+      stageName: '$default',
+      autoDeploy: true,
+    });
+
+    const integration = new apigw2.CfnIntegration(this, 'ANYIntegration', {
+      apiId: cdk.Fn.ref(api.logicalId),
+      integrationType: apigw2.HttpIntegrationType.HTTP_PROXY,
+      connectionId: paramVpcLinkId.valueAsString,
+      connectionType: apigw2.HttpConnectionType.VPC_LINK,
+      integrationMethod: 'ANY',
+      integrationUri: cloudmapService.attrArn,
+      payloadFormatVersion: '1.0',
+    });
+
+    const authorizer = new apigw2.CfnAuthorizer(this, 'Authorizer', {
+      name: `${apiName}Authorizer`,
+      apiId: cdk.Fn.ref(api.logicalId),
+      authorizerType: 'JWT',
+      jwtConfiguration: {
+        audience: [paramAppClientIdWeb && paramAppClientIdWeb.valueAsString],
+        issuer: cdk.Fn.join('', [
+          'https://cognito-idp.',
+          cdk.Aws.REGION,
+          '.amazonaws.com/',
+          paramUserPoolId && paramUserPoolId.valueAsString,
+        ]),
+      },
+      identitySource: ['$request.header.Authorization'],
+    });
+
+    authorizer.cfnOptions.condition = isAuthCondition;
+
+    new apigw2.CfnRoute(this, 'DefaultRoute', {
+      apiId: cdk.Fn.ref(api.logicalId),
+      routeKey: '$default',
+      target: cdk.Fn.join('', ['integrations/', cdk.Fn.ref(integration.logicalId)]),
+      authorizationScopes: [],
+      authorizationType: <any>cdk.Fn.conditionIf(isAuthCondition.logicalId, 'JWT', 'NONE'),
+      authorizerId: <any>cdk.Fn.conditionIf(isAuthCondition.logicalId, cdk.Fn.ref(authorizer.logicalId), ''),
+    });
+
+    new apigw2.CfnRoute(this, 'OptionsRoute', {
+      apiId: cdk.Fn.ref(api.logicalId),
+      routeKey: 'OPTIONS /{proxy+}',
+      target: cdk.Fn.join('', ['integrations/', cdk.Fn.ref(integration.logicalId)]),
+    });
+
+    //#endregion
+
+    new cdk.CfnOutput(this, 'ServiceArn', { value: cdk.Fn.ref(service.logicalId) });
+    new cdk.CfnOutput(this, 'ApiName', { value: api.name });
+    new cdk.CfnOutput(this, 'RootUrl', { value: api.attrApiEndpoint });
+
+    if (apiType === API_TYPE.GRAPHQL) {
+      new cdk.CfnOutput(this, 'GraphQLAPIEndpointOutput', { value: api.attrApiEndpoint });
+    }
+  }
+
+  getPipelineConsoleUrl(region: string) {
+    const pipelineName = this.pipeline.getPipelineName();
+    return `https://${region}.console.aws.amazon.com/codesuite/codepipeline/pipelines/${pipelineName}/view`;
+  }
+
+  toCloudFormation() {
+    prepareApp(this);
+
+    const cfn = this._toCloudFormation();
+
+    Object.keys(cfn.Parameters).forEach(k => {
+      if (k.startsWith('AssetParameters')) {
+        let value = '';
+
+        if (k.includes('Bucket')) {
+          value = this.props.deploymentBucket;
+        } else if (k.includes('VersionKey')) {
+          value = `${PIPELINE_AWAITER_ZIP}||`;
+        }
+
+        cfn.Parameters[k].Default = value;
+      }
```

Comment:
You should be able to get the cfn parameters before this in the constructor either by using where you would have the resource id to instead of iterating through this in the output cfn